language/move-binary-format/src/compatibility.rs

```diff
@@ -172,6 +182,18 @@ fn type_parameters_compatibile(
         )
 }
 
+fn type_parameters_compatibile_phantom_decl(
+    old_type_parameters: &[StructTypeParameter],
+    new_type_parameters: &[StructTypeParameter],
+) -> bool {
+    old_type_parameters.len() == new_type_parameters.len()
+        && old_type_parameters.iter().zip(new_type_parameters).all(
+            |(old_type_parameter, new_type_parameter)| {
+                old_type_parameter.is_phantom == new_type_parameter.is_phantom
```

Comment:
Same thing as the linking check. I think it is okay to add phantomness here. Because it is okay to drop it at linking

Reply:
Hopefully...